rust/cas_client/src/data_transport.rs

```diff
@@ -102,18 +108,36 @@ impl DataTransport {
     /// creates the DataTransport instance for the H2 connection using
     /// CasConnectionConfig info, and additional port
     pub async fn from_config(cas_connection_config: CasConnectionConfig) -> Result<Self> {
-        debug!(
+        info!(
             "Attempting to make HTTP connection with {}",
             cas_connection_config.endpoint
         );
-        let h2_client = Client::builder()
+        let mut builder = Client::builder(TokioExecutor::new());
+        builder
+            .timer(TokioTimer::new())
             .pool_idle_timeout(Duration::from_secs(HTTP2_POOL_IDLE_TIMEOUT_SECS))
             .http2_keep_alive_interval(Duration::from_millis(HTTP2_KEEPALIVE_MILLIS))
             .http2_initial_connection_window_size(HTTP2_WINDOW_SIZE)
             .http2_initial_stream_window_size(HTTP2_WINDOW_SIZE)
-            .http2_only(true)
-            .build_http();
-
+            .http2_only(true);
+        let connector = if let Some(root_ca) = &cas_connection_config.root_ca {
+            info!("connector with custom root ca");
+            let ca = Certificate::from_pem(root_ca.as_bytes())?;
+            let tls_connector = native_tls::TlsConnector::builder()
+                .danger_accept_invalid_certs(true)
+                .add_root_certificate(ca)
+                .build()?;
+            let tls_connector = tokio_native_tls::TlsConnector::from(tls_connector);
+            let mut sub_connector = HttpConnector::new();
+            sub_connector.enforce_http(false);
+            let mut connector = HttpsConnector::from((sub_connector, tls_connector));
+            connector.https_only(true);
+            connector
+        } else {
+            info!("default connector");
```

Comment:
when would this branch kick in during normal operations? Based on my reading, this is an error path

Reply:
This would occur if the CAS server does not send the certificate in the initiate response.

In reference to: https://github.com/xetdata/xet-core/pull/240/files#diff-2047c2e47c26df755b6edd8a9a9a04e3b259758f296d6f6748c70c4855e5591cR38, this is the behavior we would want if CAS is behind this latest version, or if we set the CAS_PROTOCOL_VERSION intentionally at 0.2.0, we should still be able to handle responses lacking the root CA.

However this is an artificial requirement I added, I can treat this as an error case instead.

Reply:
My vote is KISS and handle it as an error